aten/src/THC/generic/THCTensor.cpp

```diff
@@ -414,7 +414,7 @@ void THCTensor_(unfold)(THCState *state, THCTensor *self, THCTensor *src, int di
     }
   }
 
-  self->set_sizes_and_strides(newSize, newStride);
+  self->set_sizes_and_strides(newSize, at::optional<at::IntList>(newStride));
```

Comment:
nit: since at::optional<T> is implicitly constructible from T, I'd just make this at::IntLIst(newStride)